### PR TITLE
feat(Staking): home staking screen

### LINKED ISSUE #450: [Staking] main page
if we have no delegated rows we need to show only total balance of cards else show additional information, plus `get reward` button

All data stored at StakingMetadata table

Design: Design: https://www.figma.com/file/KoyH6PqWqfOwyIVDjnLlOk/%F0%9F%93%B1-Wallet?node-id=3076%3A52776&t=R8PhlOiWY3u155LE-4

### PR DESCRIPTION
**Pull request by gHashTag**
Fixes #450 

## Proposed Changes

- Lottie stacking animation
- Active/Inactive stacking screen
- Home stacking fix layout and implement cosmos logic
- Added a new hook that is responsible for the list of validators and for the amount of awards

### CODE REVIEW

File: src/components/home-staking/home-staking.tsx [hunk 1]
```diff
@@ -1,24 +1,115 @@
...
+        Promise.all(
+          visible
+            .filter(w => delegators.has(w.cosmosAddress))
+            .map(w =>
+              stakedValidators.map(({operator_address}) =>
+                cosmos.withdrawDelegatorReward(w.address, operator_address),
+              ),
+            )
+            .flat(),
+        ).then(() => {
```
Comment (vivalaakam) [inline src/components/home-staking/home-staking.tsx:56]:
```suggestion
        Promise.all(
          visible
            .filter(w => delegators.has(w.cosmosAddress))
            .map(w =>  cosmos.multipleWithdrawDelegatorReward(w.address, stakedValidators))
            .flat(),
        ).then(() => {
```

File: src/components/home-staking/home-staking.tsx [hunk 2]
```diff
@@ -1,24 +1,115 @@
...
+    useValidators();
+
+  const cosmos = useRef(new Cosmos(app.provider!)).current;
+  const stakingActiveRef = useRef<StakingActiveInterface>(null);
+
+  const availableSum = useMemo(() => {
+    return visible.reduce((acc, w) => acc + w.balance, 0);
+  }, [visible]);
+
+  const onWithdrawDelegatorRewards = useCallback(() => {
```
Comment (vivalaakam) [inline src/components/home-staking/home-staking.tsx:37]: should be on screen level

File: src/components/home-staking/home-staking.tsx [hunk 3]
```diff
@@ -1,24 +1,115 @@
...
+  onPressGetRewards?: () => void;
 };
-export const HomeStaking = ({onPressValidators}: StakingHomeProps) => {
+
+export const HomeStaking = ({
+  onPressValidators,
+  onPressGetRewards,
+}: StakingHomeProps) => {
+  const {visible} = useWalletsList();
+  const {loading, stakingSum, rewardsSum, unDelegationSum, stakedValidators} =
```
Comment (vivalaakam) [inline src/components/home-staking/home-staking.tsx:27]: why you getting list of validators there?

File: src/i18n.ts [hunk 4]
```diff
@@ -43,7 +43,9 @@ export enum I18N {
   accept,
   cancel,
   save,
+  staked,
```
Comment (vivalaakam) [inline src/i18n.ts:46]: use more specific key

File: src/i18n.ts [hunk 5]
```diff
@@ -43,7 +43,9 @@ export enum I18N {
...
   edit,
+  rewards,
```
Comment (vivalaakam) [inline src/i18n.ts:48]: use more specific key

File: src/screens/staking-validators.tsx [hunk 6]
```diff
@@ -1,26 +1,12 @@
-import React, {useCallback, useEffect, useRef, useState} from 'react';
+import React, {useCallback} from 'react';
```
Comment (vivalaakam) [inline src/screens/staking-validators.tsx:1]: why you move logic which should be used only on this screen into hooks?

**Review by vivalaakam — CHANGES_REQUESTED** [review]
All i

File: src/screens/staking-validators.tsx (hunk 6 above)
Comment (gHashTag) [reply src/screens/staking-validators.tsx:1]: in three places

image

File: src/screens/home-staking.tsx [hunk 7]
```diff
@@ -1,31 +1,79 @@
...
+    setData({...newData, loading: false});
+  }, []);
+
+  const onPressGetRewards = useCallback(() => {
+    const stakedValidators = StakingMetadata.getAll();
+    const rewards: Realm.Results<StakingMetadata>[] = stakedValidators.map(
+      ({validator}) => {
+        return StakingMetadata.getRewardsForValidator(validator);
+      },
+    );
```
Comment (vivalaakam) [inline src/screens/home-staking.tsx:39]:
```suggestion
    const rewards: Realm.Results<StakingMetadata>[] = stakedValidators.filtered(`type = '${StakingMetadataType.reward}'`);
```

File: src/screens/home-staking.tsx [hunk 8]
```diff
@@ -1,31 +1,79 @@
...
+
+    rewards.forEach(rewardItem => {
```
Comment (vivalaakam) [inline src/screens/home-staking.tsx:41]: Не совсем правильная логика
У тебя есть список массив со списком наград, который можно развернуть примерно как { [delegator]: validator[] } и потом по каждому из делегаторов получать все награды

Comment (vivalaakam) [reply src/screens/home-staking.tsx:41]:
```
const delegators = {}

for(const row of rewards) {
  delegators[row.delegator] = (delegators[row.delegator]  ?? []).concat(row.validator)
}

Promise.all(
  visible
             .filter(w => w.cosmosAddress in delegators)
             .map(w => cosmos.multipleWithdrawDelegatorReward(
                 w.address,
                 delegators[w.cosmosAddress],
               ))
)

```

File: src/screens/home-staking.tsx [hunk 9]
```diff
@@ -1,31 +1,79 @@
...
   const onPressValidators = useCallback(() => {
     navigation.navigate('stakingValidators');
   }, [navigation]);
 
   useEffect(() => {
-    const addressList = wallets.visible.map(w => w.cosmosAddress);
+    const newData = StakingMetadata.getSummaryInfo();
+    setData({...newData, loading: false});
+  }, []);
+
```
Comment (vivalaakam) [inline src/screens/home-staking.tsx:32]: add listener like StakingMetadata.getAll().addListener() and on change recalculate summary

Comment (vivalaakam) [reply src/screens/home-staking.tsx:32]: or recalculate summary after `cosmos.sync(addressList)` finished

**Review by vivalaakam — APPROVED** [review]

**Pull request merged**